### PR TITLE
Tweaks to branch create and rename UI

### PR DESCRIPTION
**Pull request by npostavs**
I have the vague memory that @tarsius rejected something like the branch rename change, but I can't recall why.

### CODE REVIEW

**Comment by tarsius** [comment]
> I have the vague memory that @tarsius rejected something like the branch rename change, but I can't recall why.

Looks good to me now ;-)

File: lisp/magit.el [hunk 1]
```diff
@@ -1371,6 +1371,7 @@ changes.
            (setq branch (magit-read-string-ns
                          "Branch name"
                          (and (member start (magit-list-remote-branch-names))
+                              (not (member start (magit-list-local-branch-names)))
```
Comment (tarsius) [inline lisp/magit.el]: That doesn't appear to make a difference. If start is a member of `magit-list-remote-branch-names` (e.g. "origin/feature"), then we already know that it cannot be a member of `magit-list-local-branch-names` (e.g. "feature"). Okay, except in broken cases such as when a ref `refs/heads/origin/feature` exists in a addition to `refs/remotes/origin/feature`. Am I missing something?

Comment (npostavs) [inline lisp/magit.el]: > That doesn't appear to make a difference.

Hmm, you're right, now I'm unsure why I though it did.

> I cannot reproduce that.

Oh, it's actually "when creating a new topic branch  from `origin/master`". Maybe when I tested my "fix" I entered `master`.

**Comment by npostavs** [comment]
I realized we can just use the existing `magit-revision-history` to provide the original branch name from `M-p` (like `rename-file`, not `rename-buffer`). 

Fixed my fix for the "create branch from `origin/master`" case.

Not merging immediately in case I've still left some silly mistakes.

**Comment by npostavs** [comment]
picked to `master`: c6e3293e21dc7b1bcbbcec43ad90b2ceb6aba296, 7eac94f22cddc727142d80a277a4f8452e557fc1

**Pull request closed without merging**